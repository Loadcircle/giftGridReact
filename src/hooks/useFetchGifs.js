import {useState, useEffect} from 'react'
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category)=>{

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    //esta funci贸n hace que esta fracci贸n de c贸digo solo se ejecute 1 vez (gracias al [])    
    //Los hooks no pueden ser async
    useEffect(()=>{
        getGifs(category)
            .then(resp=>{
                setState({
                    data: resp,
                    loading: false
                });
            });
    }, [category])
    //con la [ category ] le decimos que si la categoria cambia vuelva a ejecutar el c贸digo
    //en este caso nunca va a ocurrir
    

    return state; // retornamos el objeto {data:  [], loading: true}

}