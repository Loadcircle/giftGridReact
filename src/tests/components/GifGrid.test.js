const { shallow } = require("enzyme");
const { GifGrid } = require("../../components/GifGrid");
import React from 'react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom'

//esto es para simular la entrada a este archivo
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en gif grid',()=>{

    //simulamos el valor inicial de nuestro custom hook
    useFetchGifs.mockReturnValue({
        data: [],
        loading: true,        
    });
    const category = '';    
    const wrapper = shallow(<GifGrid category={category}/>)

    test('match con el esnapshot', () => {

        expect(wrapper).toMatchSnapshot();
        
    });

    test('Debe de mostrar items cuando se cargan imagenes con useFetchGifs', ()=>{

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost:8000/cosa.jpg',
            title: 'Cualquier cosa'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,        
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GridItem').length).toBe( gifs.length);

        

    });
    

});