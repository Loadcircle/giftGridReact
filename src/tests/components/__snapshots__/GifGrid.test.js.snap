// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en gif grid Debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    key=""
  />
  <div
    className="card_wrapper"
  >
    <GridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost:8000/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en gif grid match con el esnapshot 1`] = `
<Fragment>
  <h3
    key=""
  />
  <p>
    Loading...
  </p>
  <div
    className="card_wrapper"
  />
</Fragment>
`;
